include/helper_functions.php

```diff
@@ -1,15 +1,169 @@
+                                                                                                                             //this will keep only the date of the nearest class
+    $nextClasses = updateStudentInfo($nextClasses, sizeof($allStudents), 'Date', formatDate(currentDate(), 'Y-m-d H:i:s'));  //Fill out with current date those with not pending classes 
+    
+    $daysToNextCLass = calculateDaysToNextClass($nextClasses);  
+    
+}
+
+// --------------------------------------------------------------------------
+
+function updateStudentInfo($studentArray, $studentNumber, $value, $data){ 
+    $allInfo = [];
+    $temp = [];
+
+    $studentsArray = array_column($studentArray, 'Student_Id');                                                     //Get an array of ids of students with credits       
+    $studentArray = array_reverse($studentArray);                                                                   //Reverse it because I will use as stack below
+    
+    for($i = 0; $i < $studentNumber; $i++){
+
+        if(!in_array($i + 1, $studentsArray)){                                                                     //if the index is not in the array of ids
+            $array = array('Student_Id'=> $i + 1, $value=> $data);                                                 //set the id to 0 credit
+            array_push($allInfo, $array);                                                                          //push it to the array
+                    
+        }
+
+        else{                                                                                                      //if the index is in the array of ids
+            $temp = array_pop($studentArray);                                                                      //pop the last element (since it was reversed it is the first one in the original array)
+            array_push($allInfo, $temp);                                                                           //push it to the array
+        }
+    }
+
+    return $allInfo;
+}
+
+// --------------------------------------------------------------------------
+
+function removeDuplicateMultiDArray($array, $key) {                                                                 //function to remove duplicates from multidimentional arrays using a kew
+    $temp_array = [];  
+    $i = 0; 
+    $key_array = []; 
+    
+    foreach($array as $val) { 
+        if (!in_array($val[$key], $key_array)) {                                                                      //If the key is not already in the temp array stored it
+            $key_array[$i] = $val[$key]; 
+            $temp_array[$i] = $val; 
+        } 
+        $i++; 
+    } 
+    return $temp_array; 
+}
+
+// --------------------------------------------------------------------------
+
+function calculateDaysToNextClass($dates){
+    $today = new DATETIME(currentDate());
```

Comment:
For some reason (might be the format idk) it messes up with the calculation by one day if I use only new DATETIME. 
1)$today = new DATETIME(date("Y/m/d h:i:s"));            string(26) "2022-06-30 02:13:41.000000"
2)$today = new DATETIME;                                            string(26) "2022-06-30 14:14:46.198602"
So with option 2 since the format changes, if I have a class tomorrow for instance it gives me next class in 0 days, but if I use option 1 I get next class in 1 day that it is what I want.